### PR TITLE
[Lua] Add !getfame and !setfamelevel commands

### PR DESCRIPTION
**Pull request by jamesbradleym**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Adds getfame and setfamelevel commands as implemented by @CriticalXI & @paladindamarus in [ASB commit 00d8f06](https://github.com/AirSkyBoat/AirSkyBoat/commit/00d8f060aa2e7284264faf9be01f8dc31c4e209c), reformatted for LSB

## Steps to test these changes

!getfame
!setfamelevel

### CODE REVIEW

**Comment by WinterSolstice8** [comment]
please amend your commit title -- `add fame commands to LSB --author "Criticalxi"` has some extra bit at the end that shouldn't be there

**Comment by jamesbradleym** [comment]
> please amend your commit title -- `add fame commands to LSB --author "Criticalxi"` has some extra bit at the end that shouldn't be there

~~I think I'm misunderstanding this note from the [development guide](https://github.com/LandSandBoat/server/wiki/Development-Guide-Git#full-author)~~ Didn't misunderstand, just didn't do it correctly

Amended!

**Comment by TeoTwawki** [comment]
I'd very much like to see the actual fame area enumerations we already have get used, similar to how the keyitems command uses the keyitem enumeration. Its easy to miss that quest areas are enumerated because they live in the quest global still and not the enum folder yet.

**Comment by jamesbradleym** [comment]
> I'd very much like to see the actual fame area enumerations we already have get used, similar to how the keyitems command uses the keyitem enumeration. Its easy to miss that quest areas are enumerated because they live in the quest global still and not the enum folder yet.

Can do!

File: scripts/commands/getfame.lua
Comment (WinterSolstice8) [inline scripts/commands/getfame.lua:62]: Where do these come from? are they different in era vs retail?

File: scripts/commands/setfamelevel.lua
Comment (jmcmorris) [inline scripts/commands/setfamelevel.lua]: I tried using this command while testing out #6216 and discovered that `xi.quest.fame_area` does not exist on LSB and consequently the command runs into an error.

Comment (jamesbradleym) [reply scripts/commands/setfamelevel.lua]: This PR preceeded the `xi.fameArea` change, I haven't made any updates as I don't have confirmation on the `fameBaseValues`. This was migrating code from ASB work that Critical did.

Comment (paladindamarus) [reply scripts/commands/setfamelevel.lua]: I think I wrote this one originally, but yeah it was a while ago.  I can update if you want?

Comment (jamesbradleym) [reply scripts/commands/setfamelevel.lua]: Oh whoops, sorry @paladindamarus ! I had assumed from this PR: https://github.com/AirSkyBoat/AirSkyBoat/commit/00d8f060aa2e7284264faf9be01f8dc31c4e209c

Yes, feel free to 🙏🏼

Comment (paladindamarus) [reply scripts/commands/setfamelevel.lua]: Will do... And no worries, I think it was either a bulk push or maybe a completely new update when she pushed.

Comment (Xaver-DaRed) [reply scripts/commands/setfamelevel.lua]: Any movement on this?

Comment (jamesbradleym) [reply scripts/commands/setfamelevel.lua]: Last bit I heard on this is the values were taken from Topaz.

File: scripts/commands/getfame.lua
Comment (jamesbradleym) [reply scripts/commands/getfame.lua:62]: These match the values in [lua_baseentity.cpp, `getFameLevel`](https://github.com/LandSandBoat/server/blob/e55b32c476546593e089c9b246ee73109aae55ae/src/map/lua/lua_baseentity.cpp#L7119)

File: scripts/commands/setfamelevel.lua
Comment (jamesbradleym) [reply scripts/commands/setfamelevel.lua]: @Xaver-DaRed The base values match the values in [lua_baseentity.cpp, `getFameLevel`](https://github.com/LandSandBoat/server/blob/e55b32c476546593e089c9b246ee73109aae55ae/src/map/lua/lua_baseentity.cpp#L7119), if that confirmation is all that's holding up this PR then we should be good to go unless we want to centralize these values somewhere.

**Comment by jamesbradleym** [comment]
@WinterSolstice8 @Xaver-DaRed pending these checks, is there anything else that needs to be done for approval on get/set fame commands?

File: scripts/commands/setfamelevel.lua
Comment (WinterSolstice8) [reply scripts/commands/setfamelevel.lua]:
```suggestion
    for name, value in pairs(xi.fameArea) do
```

File: scripts/commands/getfame.lua
Comment (WinterSolstice8) [inline scripts/commands/getfame.lua]:
```suggestion
    for name, value in pairs(xi.fameArea) do
```

Comment (jamesbradleym) [reply scripts/commands/getfame.lua]: Done, also fixed this in other areas where xi.quest.fame_area was still being used

**Review by WinterSolstice8 — APPROVED** [review]

**Review by jmcmorris — APPROVED** [review]

**Pull request merged**
